### PR TITLE
Update Polaris tokens to expose token group types

### LINKED ISSUE #7164: Expose token group types
## Feature request summary

Export `TokenGroup`, `TokenName`, `TokenScale` type for each token group

## Rationale

So they can be imported from any component

### PR DESCRIPTION
**Pull request by aaronccasanova**
Closes #7164 

### WHAT is this pull request doing?

This PR kicks off a pattern for exposing detailed token group types and utilities:
- A `TokenGroup` type is exposed for each token group e.g. `type FontTokenGroup = {...}`
- A `TokenName` union type is exposed for each token group e.g.
  `type FontTokenName = 'font-size-1'  | 'font-size-2' | 'font-weight-medium' | 'font-weight-bold' | ...`
- A `Scale` and/or `Alias` union type is exposed for select sub-token group naming conventions (we can expand on these iteratively) e.g.
  - `type FontSizeScale = '1' | '2' | ...`
  - `type FontWeightAlias = 'medium' | 'bold' | ...`
  
Additionally, this PR improves the accuracy of a couple existing types:
- `Metadata` now represents the inferred `metadata` object
- `Tokens` now represents the inferred `metadata` object with `value` extracted

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by aveline — APPROVED** [review]
🙌

File: .changeset/little-pens-change.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-tokens': minor
+---
+
+Expose token group types
```
Comment (sam-b-rose) [inline .changeset/little-pens-change.md]: Maybe also include that we are exporting the scale and/or alias for certain token groups as well. Just for improved changelog searching 🔍

**Review by sam-b-rose — APPROVED** [review]
Looks great! Nice type improvements to `Metadata` as well. 💯

**Pull request merged**